commit 9e76ad2a27f592c1390248867391880c7efe78b3
Author: Gertjan van Wingerde <gwingerde@gmail.com>
Date:   Wed Mar 24 21:42:37 2010 +0100

    rt2x00: Disable powersaving by default in rt2500usb.
    
    Recent bug reports have shown that rt2500usb also suffers from the
    powersave problems that the PCI rt2x00 drivers suffer from.
    So disable powersaving by default for rt2500usb as well.
    
    Signed-off-by: Gertjan van Wingerde <gwingerde@gmail.com>
    Acked-by: Ivo van Doorn <IvDoorn@gmail.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

--- a/drivers/net/wireless/rt2x00/rt2500usb.c
+++ b/drivers/net/wireless/rt2x00/rt2500usb.c
@@ -1649,6 +1649,11 @@ static int rt2500usb_probe_hw_mode(struc
 	rt2x00dev->hw->wiphy->flags &= ~WIPHY_FLAG_PS_ON_BY_DEFAULT;
 
 	/*
+	 * Disable powersaving as default.
+	 */
+	rt2x00dev->hw->wiphy->flags &= ~WIPHY_FLAG_PS_ON_BY_DEFAULT;
+
+	/*
 	 * Initialize all hw fields.
 	 */
 	rt2x00dev->hw->flags =
